inc/Config/DataSource/HttpDataSource.php

```diff
@@ -44,6 +44,8 @@ final public function get_service_name(): string {
 	 * define their own validation schema.
 	 */
 	public static function preprocess_config( array $config ): array|WP_Error {
+		$config = static::perform_migration( $config );
```

Comment:
@chriszarate - the new `preprocess_config` method seems like the right place to call the migration method from within `HttpDataSource` looks like. Though if a custom implementor is used then it wouldn't be called via `from_array` so that is a downside.

What I'm curious on is, what do you think about the way the migration methods are called here?

What I haven't done yet:

- Move the version field up one level
- Update the server and UI to account for that
- Added in a sample migration. The active/version changes would really be examples of those

Reply:
> Though if a custom implementor is used then it wouldn't be called via from_array so that is a downside.

`from_array` is now marked as `final` so that shouldn't be an issue